substrate-archive/src/database/listener.rs

```diff
@@ -149,9 +150,9 @@ where
 			}
 			// collect the rest of the results, before exiting, as long as the collection completes
 			// in a reasonable amount of time
-			let timeout = smol::Timer::after(std::time::Duration::from_secs(1));
+			let timeout = smol::Timer::after(Duration::from_secs(1));
 			futures::select! {
-				_ = timeout.fuse() => {},
+				_ = FutureExt::fuse(timeout) => {},
```

Comment:
What's the reason behind this change?

Reply:
iirc won't compile because it complains that .fuse() might point to the `StreamExt`, `FutureExt` or other `fuse()` method

Reply:
Indeed. But the deeper reason the compiler can't see this escapes me.